contracts/TxRelay.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity ^0.4.8;
+
+//This contract is meant as a "singleton" forwarding contract.
+//Eventually, it will be able to forward any transaction to
+//Any contract that is built to accept it.
+
+contract TxRelay {
+
+  // Note: This is a local nonce.
+  // Different from the nonce defined w/in protocol.
+  mapping(address => uint) nonce;
+
+  /*
+   * @dev Relays normal transactions
+   * @param destination The address to relay data to
+   * @param data The bytes necessary to call the function in the destination contract.
+                 Note, the first encoded argument in data must be msg.sender's address
+   */
+  function relayTx(address destination, bytes data) payable {
+    if (!checkAddress(data, msg.sender)) throw;
+
+    //As no state is updated before, can just throw in the case of a failed call.
+    if (!destination.call.value(msg.value)(data)) throw;
+  }
+
+  /*
+   * @dev Relays meta transactions
+   * @param sigV, sigR, sigS ECDSA signature on some data to be forwarded
+   * @param data The bytes necessary to call the function in the destination contract.
+                 Note, the first encoded argument in data must be address of the signer
+   * @param claimedSender Address of the user who is having tx forwarded
+   */
+  function relayMetaTx(uint8 sigV, bytes32 sigR, bytes32 sigS,
+                       address destination, bytes data,
+                       address claimedSender) {
+
+    // relay :: nonce :: destination :: data :: relayer
+    bytes32 h = sha3(this, nonce[claimedSender], destination, data, msg.sender);
+    address addressFromSig = ecrecover(h, sigV, sigR, sigS);
+
+    nonce[claimedSender]++;
```

Comment:
No, I don't think so. It throws below if the signature ends up being incorrect, and this is reverted. For clarity, it may make sense to switch the order.